_posts/2024-09-27-learning-a-programming-language-with-generative-ai.md

```diff
@@ -0,0 +1,58 @@
+
+# The twin project application
+
+After finishing the first application, I continued to develop and learn with the actual application of the project. I implemented one of the user stories from the backlog, the purpose was to implement a data model, or in other words create database schemas and their required logic on the backend. The method was still learning by doing. I created database entities as well as controllers, services, repositories and migration scripts in .NET.  As this task was much bigger than the previous one, ChatGPT started to hallucinate and make several errors. I decided to implement additional chats. In the end I had one for the timeline of the user story, one for error handling, and one for coding with .NET.
+
+Now and then it generated not working and non-coherent code. Because of that I could not rely solely on ChatGPT but also had to google certain subjects as well as reading code documentation.
+
+In the end, however, the user story was implemented as it was supposed to be, although the implementation might not have been the best possible.
+
+# The learning experience
+
+Now and then, talking with ChatGPT might feel like having a conversation with a real person. Something that has come to my mind more than once is the Turing test. For a recap, it states that a machine can be seen as intelligent as a human, if it can engage in a conversation without being detected as a machine.  I would say that the test might now be considered as succeeded. ChatGPT can engage in a conversation by answering questions. The answers are usually grammatically correct and remind of human generated text.  And like when asking questions from a real person, the answers are not always true. The risk with a succeeding Turing test is that one might have the illusion of conversing with a human but forget that humans make mistakes. On the other hand, as it is not asking any own questions, a discussion with ChatGPT may not be a conversation in the traditional form. 
+
+![The turing tes](/img/2024-09-27-learning-a-programming-language-with-generative-ai/turing-test.png)
+
+# How to learn with GenAI
+
+Myself, I used a learning by doing approach. There are however, many ways one can learn with GenAI. Another way to learn a new programming language could be to step by step letting ChatGPT talk about the language in question, including its syntax, variables, typing, data structures and so on. I, however, would not find this approach useful as there are already online-documentation and books for most programming languages that can be used in the traditional way, and that are also reliable. Regardless of the way, there are some practices that are good to take into consideration:
```

Comment:
"typing, data structures and so on." -> "typing, data structures, and so on."

Reply:
corrected